catalyst/core/callbacks/wrapper.py

```diff
@@ -0,0 +1,298 @@
+            ignore_list.get(loader) or {}
+        )
+    else:
+        raise ValueError(
+            "'loaders' type should be one of - str, "
+            "Sequence[str], Mapping[str, int] or "
+            "Mapping[str, Sequence[int]]!"
+        )
+    return filter_fn
+
+
+def _filter_fn_from_arg(filter_fn: Union[str, FILTER_FN]) -> FILTER_FN:
+    """Check if filter function from argumets
+    can be used with filter callback.
+
+    Args:
+        filter_fn (str or Callable): filter function to check
+
+    Raises:
+        ValueError: if ``filter_fn`` is a string and can not be
+            interpreted as python code then an error will be raised
+        ValueError: if passed not callable object then will be
+            raised an error
+        ValueError: will be raised error if filter function do not
+            have three arguments
+
+    Returns:
+        filter function which accepts 3 arguments - stage (str),
+        epoch (int), loader (str) and return ``True`` if
+        need to disable callback
+    """
+    if isinstance(filter_fn, str):
+        # lambda function from string
+        try:
+            filter_fn = eval(filter_fn)  # noqa: WPS400
+        except (ValueError, SyntaxError):
+            raise ValueError(
+                "'filter_fn' should be a valid "
+                "python lambda function with "
+                "three arguments - 'stage', 'epoch' and 'loader'!"
+            )
+    if not callable(filter_fn):
+        raise ValueError("'filter_fn' should be a callable!")
+    if filter_fn.__code__.co_argcount != 3:
+        raise ValueError(
+            "Filter function should have three arguments - "
+            "'stage', 'epoch' and 'loader'!"
+        )
+    return filter_fn
+
+
+class WrapperCallback(Callback):
+    """
+    Customize callback execution on different stages, loaders and epoch.
+
+    For example, if you don't want to compute loss on a validation
+    you can ignore ``CriterionCallback``, for notebook API
+    need to wrap callback:
+
+    .. code-block:: python
+
+        from catalyst.dl import (
+            SupervisedRunner, WrapperCallback, CriterionCallback
+        )
+        runner = SupervisedRunner()
+        runner.train(
+            ...
+            callbacks=[
+                ...
+                WrapperCallback(
+                    base_callback=CriterionCallback(),
+                    loaders=["valid"]
+                ),
+                ...
+            ]
+            ...
+        )
+
+    In config API need to use ``_wrapper`` argument:
+
+    .. code-block:: yaml
+
+        callbacks_params:
+          ...
+          loss:
+            _wrapper:
+               callback: WrapperCallback
+               loaders: valid
+            callback: CriterionCallback
+          ...
+
+    """
+
+    def __init__(
+        self,
+        base_callback: Callback,
+        loaders: LOADERS = None,
+        filter_fn: Union[str, FILTER_FN] = None,
```

Comment:
would you like to add `epoch` support to run callback every `epoch` epochs? 😂

Reply:
sure